internal/model/todos.go

```diff
@@ -19,81 +19,119 @@ type TodoList struct {
 	Todos []*Todo `json:"todos"`
 }
 
-// GetTodosListByID returns slice of Todos based on the listID passed in
-func GetTodosListByID(listID string) (*TodoList, error) {
-	listStatement := `SELECT id, name FROM lists WHERE id = $1`
+// TodoListByID fetches a list and its todos
+func TodoListByID(id string) (*TodoList, error) {
 	list := &TodoList{}
 
-	err := db.QueryRow(listStatement, listID).Scan(&list.ID, &list.Name)
+	sql := "SELECT id, name FROM lists WHERE id = $1"
+	err := db.QueryRow(sql, id).Scan(&list.ID, &list.Name)
 	if err != nil {
 		return nil, err
 	}
 
-	sqlStatement := `SELECT id, list_id, todo, done FROM todos WHERE list_id = $1 ORDER BY created_at ASC;`
+	list.Todos, err = TodosByListID(id)
 
-	rows, err := db.Query(sqlStatement, listID)
+	return list, nil
+}
+
+func (tl *TodoList) validate() error {
+	if tl.Name == "" {
+		return errors.New("name empty, must be set")
+	}
+	if len(tl.Todos) == 0 {
+		return errors.New("no todos, must have at least one")
+	}
+
+	return nil
+}
+
+// Save saves a new list to the database
+func (tl *TodoList) Save() error {
+	if err := tl.validate(); err != nil {
+		return err
+	}
+	sql := `INSERT INTO lists(name) VALUES($1) RETURNING id;`
+	err := db.QueryRow(sql, tl.Name).Scan(&tl.ID)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	for rows.Next() {
-		todo := &Todo{}
-		err := rows.Scan(&todo.ID, &todo.ListID, &todo.Todo, &todo.Done)
+	for _, t := range tl.Todos {
+		t.ListID = tl.ID
+		err := t.Save()
 		if err != nil {
-			return nil, err
+			return err
 		}
-		list.Todos = append(list.Todos, todo)
 	}
-	return list, nil
+
+	return nil
 }
 
-// CreateTodos inserts todos into the todos table
-func CreateTodos(name string, todos []string) ([]*Todo, error) {
-	var newTodos []*Todo
-	listID, err := createList(name)
+// TodoByID fetches a single todo given a listID and todoID
+func TodoByID(listID, todoID string) (*Todo, error) {
+	t := &Todo{}
+	sql := `SELECT id, list_id, todo, done FROM todos WHERE list_id = $1 AND id = $2 ORDER BY created_at ASC;`
+	err := db.QueryRow(sql, listID, todoID).Scan(&t.ID, &t.ListID, &t.Todo, &t.Done)
+	if err != nil {
+		return nil, err
+	}
+
+	return t, nil
+}
```

Comment:
A new func that's useful for loading a Todo, like when we toggle the done checkmark. We'll see where we call this later.